src/main/java/Duke.java

```diff
@@ -1,10 +1,214 @@
+        }
+
+        // Read input such that it is not "bye"
+        while(!input.equals("bye")) {
+
+            // Split user input to check for dynamism
+            String originalInput = input;
+            String[] splitInput = input.split(" ");
+            input = splitInput[0];
+            int numTasks = Task.getTasksArray().size();
+
+            // Switch to check input
+            switch(input){
+
+            // Lists all tasks using the Task class printAllTask() method
+            case "list": {
+                try {
+                    if(numTasks == 0) { 
+                        throw new DukeExceptions.noTasksException("Ye have no tasks yet, ye lazy buccaneer!");
+                    }
+                    System.out.println(BARRIER + "\n");
+                    Task.printAllTasks();
+                    System.out.println(BARRIER + "\n");
+                } catch (DukeExceptions.noTasksException e) {
+                    System.out.println(BARRIER + "\n");
+                } finally {
+                    break;
+                }
+            }
+
+            // Marks a task as complete
+            case "mark": {
+                try {
+                    if (!isValidInput(splitInput[1])) {
+                        throw new DukeExceptions.invalidNumberException(BARRIER + "\n\nBlast! That isn't a valid number lad!\n" + BARRIER + "\n");
+                    }
+                    int index = Integer.parseInt(splitInput[1]) - 1;
+                    Task.getTasksArray().get(index).markAsComplete();
+                } finally {
+                    break;
+                }
+            }
+
+            // Unmarks a previous completed task
+            case "unmark": {
+                try {
+                    if (!isValidInput(splitInput[1])) {
+                        throw new DukeExceptions.invalidNumberException(BARRIER + "\n\nBlast! That isn't a valid number lad!\n" + BARRIER + "\n");
+                    }
+                    int index = Integer.parseInt(splitInput[1]) - 1;
+                    Task.getTasksArray().get(index).unmarkAsComplete();
+                } finally {
+                    break;
+                }
+            }
+
+            // Prints a list of commands and description
+            case "help": {
+                printHelpList();
+                break;
+            }
+
+            // Adds a new ToDo task
+            case "todo": {
+                System.out.println(BARRIER + "\n");
+                Todo temp = new Todo(originalInput);
+                System.out.println("added: " + temp.printTask());
+                printListLength();
+                System.out.println(BARRIER + "\n");
+                break;
+            }
+
+            // Adds a new Deadline task
+            case "deadline": {
+                // Command input starts after '/'
+                int commandStart = originalInput.indexOf("/");
```

Comment:
I observed that you use the `/` character several times as a delimiter. Perhaps you can consider assigning this value to a variable and referencing it as such in order to minimise magic literals within your code.